projects/igniteui-angular/src/lib/time-picker/time-picker.directives.ts

```diff
@@ -166,55 +170,23 @@ export class IgxItemListDirective {
 
         const delta = event.deltaY;
         if (delta !== 0) {
-            switch (this.type) {
-                case 'hourList': {
-                    this.timePicker.nextHour(delta);
-                    break;
-                }
-                case 'minuteList': {
-                    this.timePicker.nextMinute(delta);
-                    break;
-                }
-                case 'secondsList': {
-                    this.timePicker.nextSeconds(delta);
-                    break;
-                }
-                case 'ampmList': {
-                    this.timePicker.nextAmPm(delta);
-                    break;
-                }
-            }
+            this.nextItem(delta);
         }
     }
 
     /**
-     * @hidden
+     * @hidden @internal
      */
-    @HostListener('panmove', ['$event'])
-    public onPanMove(event) {
-        const delta = event.deltaY < 0 ? 1 : event.deltaY > 0 ? -1 : 0;
-        if (delta !== 0) {
-            switch (this.type) {
-                case 'hourList': {
-                    this.timePicker.nextHour(delta);
-                    break;
-                }
-                case 'minuteList': {
-                    this.timePicker.nextMinute(delta);
-                    break;
-                }
-                case 'secondsList': {
-                    this.timePicker.nextSeconds(delta);
-                    break;
-                }
-                case 'ampmList': {
-                    this.timePicker.nextAmPm(delta);
-                    break;
-                }
-            }
-        }
+    public ngOnInit() {
+        const hammerOptions: HammerOptions = { recognizers: [[Hammer.Pan, { direction: Hammer.DIRECTION_VERTICAL, threshold: 5 }]] };
+        this.touchManager.addEventListener(this.elementRef.nativeElement, 'pan', this.onPanMove, hammerOptions);
```

Comment:
I think it's a good idea to destroy the `touchManager`  in `ngOnDestroy` since it may cause memory leaks.